test/useConfirm.test.js

```diff
@@ -435,6 +435,31 @@ describe("useConfirm", () => {
     });
   });
 
+  test("does not close the modal when another component with useConfirm is unmounted and closeOnParentUnmount is false", async () => {
+    const ParentComponent = ({}) => {
+      const [alive, setAlive] = useState(true);
+
+      return (
+        <ConfirmProvider>
+          {alive && <DeleteButton confirmOptions={{ closeOnParentUnmount: false }} text="Delete 1" />}
+          <DeleteButton confirmOptions={{ closeOnParentUnmount: false }} text="Delete 2" />
+          <button onClick={() => setAlive(false)}>Unmount child</button>
+        </ConfirmProvider>
+      );
+    };
+
+    const { getByText, queryByText } = render(<ParentComponent />);
+
+    fireEvent.click(getByText("Delete 2"));
+    expect(queryByText("Are you sure?")).toBeTruthy();
+
+    // Remove the first <DeleteButton /> from the tree
+    fireEvent.click(getByText("Unmount child"));
+
+    fireEvent.click(getByText("Ok"));
+    expect(deleteConfirmed).not.toHaveBeenCalled();
+  });
```

Comment:
We actually want to assert that the modal confirmation still works.

```suggestion
    test("does not close the modal when opening component is unmounted when closeOnParentUnmount is false", async () => {
      const ParentComponent = ({}) => {
        const [alive, setAlive] = useState(true);

        return (
          <ConfirmProvider>
            {alive && <DeleteButton confirmOptions={{ closeOnParentUnmount: false }} />}
            <button onClick={() => setAlive(false)}>Unmount child</button>
          </ConfirmProvider>
        );
      };

      const { getByText, queryByText } = render(<ParentComponent />);

      fireEvent.click(getByText("Delete"));
      expect(queryByText("Are you sure?")).toBeTruthy();

      // Remove <DeleteButton /> from the tree
      fireEvent.click(getByText("Unmount child"));

      // The modal is still functional
      fireEvent.click(getByText("Ok"));
      await waitForElementToBeRemoved(() => queryByText("Are you sure?"));
      expect(deleteConfirmed).toHaveBeenCalled();
    });
```